src/core/namespace.c

```diff
@@ -1128,9 +1128,23 @@ static int mount_procfs(const MountEntry *m, const NamespaceInfo *ns_info) {
                 r = path_is_mount_point(entry_path, NULL, 0);
                 if (r < 0)
                         return log_debug_errno(r, "Unable to determine whether /proc is already mounted: %m");
-                if (r == 0)
-                        /* /proc is not mounted. Propagate the original error code. */
-                        return -EPERM;
+                if (r == 0) {
+                        r = path_is_mount_point("/proc", NULL, 0);
+                        if (r < 0)
+                                return log_debug_errno(r, "Unable to determine whether /proc is already mounted: %m");
+                        if (r == 0)
+                                /* /proc is not mounted, neither at the target nor on the host. Propagate
+                                 * the original error code. */
+                                return -EPERM;
```

Comment:
hmm, why this test? /proc/ is definitely going to be mounted on the host, we generally don't need to check that, do we?

Is there are reason for adding this, or just extra safety? I'd really drop this check.

Reply:
Just an extra check, so that if access is denied in some way and the check fails we get a (hopefully) more helpful log_debug above. But I can remove it if that's the preference.

Reply:
i'd drop it, yeah. just seems unlikely to be hit, at least i don't see how it could ever happen